File: sol/2654/2654.cpp
#include <numeric>

struct Solution {
    int minOperations(vector<int>& nums) {
        int n = nums.size(), nonone = 0;
        for (int x : nums){
            if (x != 1){
                ++nonone;
            }
        }
        if (nonone != n){
            return nonone;
        }
        int minLen = n, g = nums[0];
        for (int j = 1; j < n; ++j){
            g = gcd(g, nums[j]);
            if (g == 1){
                minLen = j + 1;
                break;
            }
        }
        if (g != 1){
            return -1;
        }
        for (int i = 1; i < n; ++i){
            g = nums[i];
            for (int j = i + 1; j < n; ++j){
                g = gcd(g, nums[j]);
                if (g == 1){
                    minLen = min(minLen, j - i + 1);
                    break;
                }
            }
        }
        return minLen + n - 2;
    }
};

File: sol/2654/2654.py
from math import gcd

class Solution:
    def minOperations(self, nums: List[int]) -> int:
        n = len(nums)
        if 1 in nums:
            return n - nums.count(1)
        minLen = n
        g = nums[0]
        for j in range(1, n):
            g = gcd(g, nums[j])
            if g == 1:
                minLen = j + 1
                break
        if g != 1:
            return -1
        for i in range(1, n):
            g = nums[i]
            for j in range(i + 1, n):
                g = gcd(g, nums[j])
                if g == 1:
                    minLen = min(minLen, j + 1 - i)
                    break
        return minLen + n - 2
